### PR TITLE
Various Architecture Decision Records regarding our ElasticSearch usage

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
See https://github.com/apache/james-project/pull/171

File: src/adr/0010-enable-elasticsearch-routing.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# 10 Disable ElasticSearch dynamic mapping
```
Comment (MichaelBailly) [inline src/adr/0010-enable-elasticsearch-routing.md]: wrong title

File: src/adr/0011-remove-elasticsearch-document-source.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# 11. Disable ElasticSearch dynamic mapping
```
Comment (MichaelBailly) [inline src/adr/0011-remove-elasticsearch-document-source.md]: wrong title

File: src/adr/0010-enable-elasticsearch-routing.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+Date: 2019-10-17
+
+## Status
+
+Proposed
```
Comment (blackheaven) [inline src/adr/0010-enable-elasticsearch-routing.md]: Wrong status?

File: src/adr/0011-remove-elasticsearch-document-source.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+## Consequences
+
+We notice a 20% space reduction of data stored on ElasticSearch.
+
+However, patch updates can no longer be performed upon flags updates. Upon flag update we need to fully read the mail 
+content, then mime-parse it, potentially html parse it, extract attachment content again and finally index again the full 
+document.
+
+Without _source field, flags update is two time slower, 99 percentile 4 time slower, and this impact negatively other
```
Comment (blackheaven) [inline src/adr/0011-remove-elasticsearch-document-source.md]: lack of quote

File: src/adr/0009-disable-elasticsearch-dynamic-mapping.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
+# 9. Disable ElasticSearch dynamic mapping
+
+Date: 2019-10-10
+## Status
+
+Accepted (lazy consensus)
+
+## Context
+
+We rely on dynamic to expose our  mail headers as a JSON map. Dynamic mapping is enabled for adding not yet encountered headers in the mapping.
```
Comment (remk) [inline src/adr/0009-disable-elasticsearch-dynamic-mapping.md]: s/dynamic/dynamic mapping
Two spaces between `our` and `mail`

File: src/adr/0009-disable-elasticsearch-dynamic-mapping.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+ - Field type 'guess' can be wrong, leading to subsequent headers omissions [1]
+ - Document indexation needs to be paused at the index level during mapping changes to avoid concurrent changes, impacting negatively performance.
+
+## Decision
+
+Rely on nested objects to represent mail headers within a mapping
+
+## Consequences
+
+The index needs to be re-created. Document reIndexation is needs.
```
Comment (remk) [inline src/adr/0009-disable-elasticsearch-dynamic-mapping.md]: s/is needs/is needed

Comment (chibenwa) [reply src/adr/0009-disable-elasticsearch-dynamic-mapping.md]: Can you please stop commenting closed PRs?

**Pull request closed without merging**